Fix header/library locations for fltk, respect LDFLAGS, and allow the ebuild to
handle docs install.

--- mup-6.6/makefile
+++ mup-6.6/makefile
@@ -80,12 +80,12 @@
 
 # You can change this if your X libraries and headers are somewhere else
 # (like /usr/X11R6).
-X_LOCATION = /usr/X11
+X_LOCATION = /usr
 
 # If you installed fltk somewhere other than $(X_LOCATION)/lib
 # set this to find them, as in
 # FLTK_LIB_LOCATION = -L/usr/local/lib
-FLTK_LIB_LOCATION =
+FLTK_LIB_LOCATION = -L/usr/lib/fltk
 
 # The X libraries to link with Mupmate.
 # Depending on how fltk was compiled, you may be able to omit
@@ -99,7 +99,7 @@
 #FLTK_VERSION=FLTK_1_1
 
 # The FLTK header files are under $(FLTK_INCLUDE)/FL
-FLTK_INCLUDE = $(X_LOCATION)/include
+FLTK_INCLUDE = $(X_LOCATION)/include/fltk
 
 # You can use fltk_jpeg or jpeg library, whichever you have
 # or depending on how fltk was compiled, you may not need it at all
@@ -251,25 +251,25 @@
 	flex -t extras/lex.l > mup/lex.c
 
 mup/mup: $(MUP_HDRS) $(MUP_SRC)
-	$(CCOMPILER) $(CFLAGS) -o $@ $(MUP_SRC) -lm
+	$(CCOMPILER) $(CFLAGS) $(LDFLAGS) -o $@ $(MUP_SRC) -lm
 
 mupdisp/mupdisp: $(MUPDISP_HDRS) $(MUPDISP_BITMAPS) $(MUPDISP_SRC)
-	$(CCOMPILER) $(CFLAGS) -L$(X_LOCATION)/lib -o $@ -DNO_VGA_LIB $(MUPDISP_SRC) -lX11
+	$(CCOMPILER) $(CFLAGS) $(LDFLAGS) -L$(X_LOCATION)/lib -o $@ -DNO_VGA_LIB $(MUPDISP_SRC) -lX11
 	# For Linux console mode support, comment out the previous line
 	# and uncomment the following line
 	# $(CCOMPILER) $(CFLAGS) -L$(X_LOCATION)/lib -o $@ $(MUPDISP_SRC) -lvga -lX11 -lm
 
 mkmupfnt/mkmupfnt: $(MKMUPFNT_SRC)
-	$(CCOMPILER) $(CFLAGS) -o $@ $(MKMUPFNT_SRC)
+	$(CCOMPILER) $(CFLAGS) $(LDFLAGS) -o $@ $(MKMUPFNT_SRC)
 
 mupmate/mupmate: $(MUPMATE_SRC) $(MUPMATE_HDRS) $(MUPMATE_OTHER_FILES)
-	$(CPPCOMPILER) $(CFLAGS) -o $@ $(MUPMATE_SRC) \
+	$(CPPCOMPILER) $(CXXFLAGS) $(LDFLAGS) -o $@ $(MUPMATE_SRC) \
 	$(FLTK_VERSION) -I$(FLTK_INCLUDE) -Imup -L$(X_LOCATION)/lib \
 	$(FLTK_LIB_LOCATION) -lfltk -lfltk_images $(X_LIBS) \
 	$(JPEGLIB) $(PNGLIB) $(ZLIB) -lm
 
 install:	all
-	mkdir -p $(BINDIR) $(MANDIR) $(DOCDIR)/uguide $(ICONDIR) $(LIBDIR)
+	mkdir -p $(BINDIR) $(MANDIR) $(ICONDIR) $(LIBDIR)
 	cp mup/mup $(BINDIR)/mup
 	cp mkmupfnt/mkmupfnt $(BINDIR)/mkmupfnt
 	cp mupdisp/mupdisp $(BINDIR)/mupdisp
@@ -278,8 +278,6 @@
 	cp mupmate/mupmate $(BINDIR)/mupmate
 	cp mupprnt $(BINDIR)/mupprnt
 	cp docs/*.1 $(MANDIR)
-	cp docs/README* docs/*.txt docs/*.ps docs/*.mup $(DOCDIR)
-	cp docs/uguide/* $(DOCDIR)/uguide
 	cp $(MUPMATE_OTHER_FILES) $(ICONDIR)
 	cp $(MUP_LIB_FILES) $(LIBDIR)
 
